pydantic_ai_slim/pydantic_ai/models/anthropic.py

```diff
@@ -718,10 +719,27 @@ async def _map_message(  # noqa: C901
                             else:
                                 user_content_params.append(content)
                     elif isinstance(request_part, ToolReturnPart):
+                        tool_result_content: list[BetaContentBlockParam] = []
+                        for part in request_part.model_response_parts():
+                            if isinstance(part, str):
+                                if part:
+                                    tool_result_content.append(BetaTextBlockParam(text=part, type='text'))
+                            elif isinstance(part, (FileUrl, BinaryContent)):
+                                user_prompt = UserPromptPart(content=[part])  # pyright: ignore[reportArgumentType]
+                                async for block in self._map_user_prompt(user_prompt):
+                                    if not isinstance(block, CachePoint):
+                                        tool_result_content.append(block)
```

Comment:
It doesn't look like all the same type are supported as tool results, i.e. images yes but not documents. So we may need a hybrid approach of directly returned results + separate message.